File: app_mantenimiento/lib/services/reminder_service.dart
import '../config/api_config.dart';
import '../models/reminder/reminder_response.dart';
import 'api_service.dart';

class ReminderService {
  Future<List<ReminderResponse>> getUserReminders() async {
    final data = await ApiService.get(ApiConfig.reminders);
    return ApiPayload.decodeList(data).map((e) => ReminderResponse.fromJson(e)).toList();
  }

  Future<List<ReminderResponse>> getDueReminders() async {
    final data = await ApiService.get(ApiConfig.remindersDue);
    return ApiPayload.decodeList(data).map((e) => ReminderResponse.fromJson(e)).toList();
  }

  Future<List<ReminderResponse>> getRemindersByVehicle(int vehicleId) async {
    final data =
        await ApiService.get(ApiConfig.remindersByVehicle(vehicleId));
    return ApiPayload.decodeList(data).map((e) => ReminderResponse.fromJson(e)).toList();
  }

  Future<void> deleteReminder(int id) async {
    await ApiService.delete(ApiConfig.reminderById(id));
  }

  Future<void> toggleReminder(int id) async {
    await ApiService.patch(ApiConfig.reminderToggle(id));
  }
}

File: frontend/src/app/core/services/reminder.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { ReminderResponse } from '../../models/reminder/reminder-response';

@Injectable({ providedIn: 'root' })
export class ReminderService {
  private apiUrl = environment.apiUrl;

  constructor(private http: HttpClient) {}

  getUserReminders(): Observable<ReminderResponse[]> {
    return this.http.get<ReminderResponse[]>(`${this.apiUrl}/reminders`);
  }

  getDueReminders(): Observable<ReminderResponse[]> {
    return this.http.get<ReminderResponse[]>(`${this.apiUrl}/reminders/due`);
  }

  getRemindersByVehicle(vehicleId: number): Observable<ReminderResponse[]> {
    return this.http.get<ReminderResponse[]>(
      `${this.apiUrl}/reminders/vehicle/${vehicleId}`
    );
  }

  deleteReminder(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/reminders/${id}`);
  }
}
